py/terra_widgets/workspace_metadata.py

```diff
@@ -0,0 +1,79 @@
+"""Methods to obtain workspace metadata for the current user in various formats."""
+
+import json
+import os
+from typing import Dict
+
+import firecloud.api as fapi
+from IPython import get_ipython
+
+AOU_DOMAIN = '@researchallofus.org'
+EDIT_ACCESS_LEVELS = ['WRITER', 'OWNER', 'PROJECT_OWNER']
+
+
+class WorkspaceMetadata:
+  """Encapsulate all logic for obtaining workspace metadata."""
+
+  def __init__(self):
+    self.user = os.getenv('OWNER_EMAIL')
+    self.terra_workspaces = fapi.list_workspaces().json()
+    if self.user.endswith(AOU_DOMAIN):
+      # Use the All of Us API to get the human-readable workspace names. For All of Us workspaces,
+      # the Terra workspace metadata the workspace names are actually the AoU workspace ids.
+      aou_response = get_ipython().getoutput('''curl -H "Content-Type: application/json" \
+          -H "Authorization: Bearer $(gcloud auth print-access-token)" \
+          "https://api.workbench.researchallofus.org/v1/workspaces" 2>/dev/null | jq .''')
```

Comment:
Note: this will only work in production. We don't have a great way of plumbing the APi host through to your code, so I don't think there's much we can do about that for now. Will include this in above discussion.

Reply:
any differences in environment variable values that might indicate dev vs. prod?

Reply:
yes - it could be inferred very indirectly based on the project name. There are 6 AoU environments so it would be a painfully brittle mapping to maintain. Perhaps for now just read a non-existent env var (e.g. `RW_API_BASE_URL`), and fallback to prod if unset. This would allow us to test in lower environments if needed by setting the env var. May need a similar thing to specify the Firecloud environment - I'm not sure if there's already an env var to help for this.

Reply:
added that env var check